nautobot/extras/models/jobs.py

```diff
@@ -571,6 +578,16 @@ def display(self):
         workers = "worker" if worker_count == 1 else "workers"
         return f"{self.queue_type}: {self.name} ({worker_count} {workers})"
 
+    def clean(self):
+        super().clean()
+        # Validate that only kubernetes job queue can be linked to an external integration object.
+        if self.external_integration and self.queue_type != JobQueueTypeChoices.TYPE_KUBERNETES:
```

Comment:
I'm wondering if we want the converse as well - if TYPE_KUBERNETES, should an external_integration be required?